src/set.js

```diff
@@ -46,11 +45,15 @@ Sk.builtin.set.prototype["$r"] = function () {
     for (it = this.tp$iter(), i = it.tp$iternext(); i !== undefined; i = it.tp$iternext()) {
         ret.push(Sk.misceval.objectRepr(i).v);
     }
-    return new Sk.builtin.str("set([" + ret.join(", ") + "])");
+    if(Sk.python3) {
+        return new Sk.builtin.str("{" + ret.join(", ") + "}");
+    } else {
+        return new Sk.builtin.str("set([" + ret.join(", ") + "])");
+    }
 };
 Sk.builtin.set.prototype.tp$getattr = Sk.builtin.object.prototype.GenericGetAttr;
 // todo; you can't hash a set() -- what should this be?
-Sk.builtin.set.prototype.tp$hash = Sk.builtin.object.prototype.HashNotImplemented;
+Sk.builtin.set.prototype.tp$hash = Sk.builtin.none.none$;
```

Comment:
That is correct, I've added a new case in the ```hash``` builtin, but it doesn't work anymore. I investigate. Setting it to ```none``` and handling this in the builtin is the cpython behavior.